src/tp/tp_segmenter.cpp

```
@@ -102,7 +102,7 @@ TpResult TpSegmenter::create_multi_segments(const Message& message,
         header.resize(16);  // Keep only header (16 bytes)
 
         // Add first part of payload (accounting for TP header)
-        size_t first_payload_size = std::min(static_cast<size_t>(config_.max_segment_size - 16 - 4),
+        size_t const first_payload_size = std::min(static_cast<size_t>(config_.max_segment_size - 16 - 4),
```

Cppcheck (2.20.0)

[style] The function 'get_client_id' is never used.

(unusedFunction)